manager/controllers/app/dataset/data_info.go

```diff
@@ -0,0 +1,52 @@
+// Copyright 2020 IBM Corp.
+// SPDX-License-Identifier: Apache-2.0
+
+package dataset
+
+import (
+	"errors"
+
+	"fybrik.io/fybrik/pkg/serde"
+
+	app "fybrik.io/fybrik/manager/apis/app/v1alpha1"
+	"fybrik.io/fybrik/manager/controllers/utils"
+	pb "fybrik.io/fybrik/pkg/connectors/protobuf"
+)
+
+// DataDetails is the asset metadata received from the catalog connector
+// This structure is in use by the manager and other components, such as policy manager and config policies evaluator
+type DataDetails struct {
+	// Name of the asset
+	Name string
+	// Interface is the protocol and format
+	Interface app.InterfaceDetails
+	// Geography is the geo-location of the asset
+	Geography string
+	// Connection is the connection details in raw format as received from the connector
+	Connection serde.Arbitrary
+	// Metadata
+	Metadata *pb.DatasetMetadata
+}
+
+// Transforms a CatalogDatasetInfo into a DataDetails struct
+// TODO Think about getting rid of one or the other and reuse
```

Comment:
Agree.  We shouldn't be doing such transforms.  There should be a single structure used (and validated via the taxonomy) throughout.

Reply:
Please open an issue for this